src/components/FormFields/ParameterFields.js

```diff
@@ -6,13 +6,33 @@ import { hooks } from '../Store'
 import { formatToString } from './formatters'
 import SkipTableTypesField from './SkipTableTypesField'
 import LabeledOptionsField from './LabeledOptionsField'
+import DataIntegrityChecksField from './DataIntegrityChecksField'
+import DataIntegrityReportTypeField from './DataIntegrityReportTypeField'
 import styles from './ParameterFields.module.css'
 
 const { Field } = ReactFinalForm
 
 // The key under which the parameters will be sent to the backend
 const FIELD_NAME = 'jobParameters'
 
+const JOB_TYPES = {
+    DATA_INTEGRITY: 'DATA_INTEGRITY',
+}
+
+const getCustomComponent = (jobType, parameterName) => {
+    if (jobType === JOB_TYPES.DATA_INTEGRITY && parameterName === 'checks') {
+        return DataIntegrityChecksField
+    } else if (
+        jobType === JOB_TYPES.DATA_INTEGRITY &&
+        parameterName === 'type'
+    ) {
+        return DataIntegrityReportTypeField
+    } else if (parameterName === 'skipTableTypes') {
+        return SkipTableTypesField
+    }
+    return null
+}
+
 // Renders all parameters for a given jobtype
 const ParameterFields = ({ jobType }) => {
```

Comment:
I think these changes go somewhat against the grain of the design of the app (just like the skipTableTypes parameter). The app is meant to respond to the `klass` types (however ugly that is) and render the appropriate fields.

I see that the backend returns a `java.util.Set` type for the `Checks` parameter. We used to have a field for that type in the app that was removed because it was unused. Would it be sufficient to assume that the Set type indicates the transfer component? Or is this control so specific that it can't be solved in a generic manner? Because that would align with the existing design and hopefully then also generalize to other parameters that might need a transfer component.

Also, the `org.hisp.dhis.scheduling.parameters.DataIntegrityJobParameters$DataIntegrityReportType` klass for the `Type` field seems very inconsistent with the other klass fields. This seems to just render a single select. For other parameters we use a `java.util.List` for that. Is there a reason that the backend doesn't just return a `java.util.List` there?

Reply:
I'm not saying that I like the current design of dynamically responding to the klass fields btw. I think we should move to fixed forms like the usage-analytics-app. But until we do that I think it can get confusing if we start mixing approaches.

Reply:
> Should probably create a generic field component for 'java.lang.Set', however we still need to override the TransferOption due to the label not supporting a react-component. I plan to update this in UI though.
>
> Note that the https://debug.dhis2.org/dev/api/dataIntegrity.json. endpoint contains properties like description, recommendation and introduction.
>
> These are very inconsistent, and most jobs do not have all these properties - and description seem to just be a human readable name. After a brief discussion with Joe, we decided a simple MVP is good enough for now. So we opted to just show the name and severity.

I think this explains most of it. So basically the transfer component needs specifics that can't really be expressed with just the `Set` klass type. Seems that also here my questions should be directed to backend.

Reply:
I know that you resolved this, but I will just add my thoughts here :)

> I think this explains most of it. So basically the transfer component needs specifics that can't really be expressed with just the Set klass type. Seems that also here my questions should be directed to backend.

Exactly, I've also looked into a generic "transfer"-component in a separate branch, that we can use for an upcoming feature. I also tried to look into adding pagination to this, but that is very hacky at the moment due to the `runtime` not supporting dynamic queries. 

> I think we should move to fixed forms like the usage-analytics-app.

This is something we can discuss, but there's definitely some advantages to have some dynamic rendering of the forms. This is something that will come up in regards to the Maintenance refactor, so we will probably discuss it a lot then. 

In regards to your points about this being "against the design of the app"; In any situation where we implement something dynamic from the server there has to be ways to override them. Of course, in an ideal world we would get all we need from the server and just render it. However, we cannot expect to ever get enough information to always be able to rely on that, and there will always be anomalies - eg; how this transfer component uses custom rendering of the options. And I don't think this is a bad thing. The Maintenance app gets a lot of flak, but some of the great parts is how easy it is to add new fields to due the dynamic resolution of fields. 
Of course if we move to more fixed fields, all this doesn't really matter. In the meantime there's not really an alternative here. It doesn't really have to be confusing either, as we could integrate the `custom`-fields solution better, but I didn't want to change too much. 

> org.hisp.dhis.scheduling.parameters.DataIntegrityJobParameters$DataIntegrityReportType

I was a bit confused by this as well, but I felt that I had already bothered the backend dev enough already, and it was after soft freeze. I think it has to do with the `constants`-array that is included in the payload. We are also using the `name` and `jobType`-props to resolve the field regardless.